lib/Service/CouponService.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Stripe\Service;
+
+class CouponService extends AbstractService
+{
+    public function basePath()
+    {
+        return "/v1/coupons";
+    }
+
+    /**
+     * List all coupons.
+     *
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\Coupon
```

Comment:
it should be a list of coupons right?

Reply:
Nice catch. Yes, the return type should be `\Stripe\Collection`.

Reply:
Can it be special-cased for Coupons? so that `$ret->data` is "typed" in IDEs?

Reply:
Unfortunately AFAIK no, PHPDoc has no support for "generics".